csm/core/services/session/session_factory.py

```diff
@@ -0,0 +1,178 @@
+
+    async def get_all(self):
+        return list(self._stg.values())
+
+    async def store(self, session: Session) -> None:
+        self._stg[session.session_id] = session
+
+class Database:
+    def __init__(self, storage: DataBaseProvider):
+        self.storage = storage
+
+    async def convert_model_to_session(self, session_model_list):
+        session_list = []
+        for model in session_model_list:
+            session = Session(model._session_id,
+                              model._expiry_time,
+                              SessionCredentials(model._user_id),
+                              PermissionSet(model._permission))
+            session_list.append(session)
+        return session_list
+
+    async def convert_session_to_model(self, session:Session):
+        sessionModel = SessionModel.instantiate_session(session._session_id, session._expiry_time,
+                                                        session._credentials._user_id,
+                                                        session._permissions._items)
+        return sessionModel
+
+    async def delete(self, session_id: Session.Id) -> None:
+        await self.storage(SessionModel).delete(Compare(SessionModel._session_id, '=', session_id))
+
+    async def get(self, session_id: Session.Id) -> Optional[Session]:
+        query = Query().filter_by(Compare(SessionModel._session_id, '=', session_id))
+        session__model_list = await self.storage(SessionModel).get(query)
+        """
+        Storage get() :
+        :param query: session id
+        :return: empty list or list with session model object which satisfy the passed query condition
+        """
+        session_list = await self.convert_model_to_session(session__model_list)
+        if session_list:
+            return session_list[0]
+        else:
+            return None
+
+    async def get_all(self):
+        #TODO :- need to verify
+        # Convert SessionModel to Session
+        query = Query()
+        session__model_list = await self.storage(SessionModel).get(query)
+        session_list = await self.convert_model_to_session(session__model_list)
+        return session_list
+
+    async def store(self, session: Session) -> None:
+        # Convert session to SessionModel
+        sessionModel = await self.convert_session_to_model(session)
+        await self.storage(SessionModel).store(sessionModel)
+
+class SessionFactory:
+    @staticmethod
+    def get_session(storage: DataBaseProvider=None):
+        backend = Conf.get(const.CSM_GLOBAL_INDEX, const.SESSION_BACKEND_KEY)
+        #TODO : Nedd to discuss
+        session_backend_map = {
+            const.LOCAL: const.IN_MEMORY,
+            const.PERSISTENT: const.DB
+        }
+        session_backend = session_backend_map[backend]
+        if session_backend == const.DB:
+            if storage is None:
+                #TODO : Verify the error
+                raise CsmError("Persistent Database is NULL")
```

Comment:
CsmInternalError would be an appropriate error here.

Reply:
ok I will update this

Reply:
please modify the error message too.

Reply:
updated